catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/util/EndpointUtil.java

```diff
@@ -164,6 +177,46 @@ public Metacard getMetacard(String id)
         return result.getMetacard();
     }
 
+    /**
+     * Returns all associated metacards given two relatable attributes.
+     *
+     * @param primaryFilterAttribute   The primary attribute to search against
+     * @param secondaryFilterAttribute The secondary attribute to search against
+     * @param primaryAttributeValue    A primary attribute value
+     * @param secondaryAttributeValue  A secondary attribute value
+     * @return A set of metacards, filtered by the given attributes and it's values
+     * @throws UnsupportedQueryException  Error
+     * @throws SourceUnavailableException Error
+     * @throws FederationException        Error
+     */
+    public List<Metacard> getAssociatedMetacardsByTwoAttributes(String primaryFilterAttribute,
+            String secondaryFilterAttribute, String primaryAttributeValue,
+            String secondaryAttributeValue)
+            throws UnsupportedQueryException, SourceUnavailableException, FederationException {
+        LinkedList<Metacard> resultList = new LinkedList<>();
+        Filter idFilter = filterBuilder.attribute(primaryFilterAttribute)
+                .is()
+                .equalTo()
+                .text(primaryAttributeValue);
+        Filter tagsFilter = filterBuilder.attribute(secondaryFilterAttribute)
+                .is()
+                .equalTo()
+                .text(secondaryAttributeValue);
+        Filter filter = filterBuilder.allOf(idFilter, tagsFilter);
+        QueryResponse queryResponse = catalogFramework.query(new QueryRequestImpl(new QueryImpl(
+                filter), false));
```

Comment:
❗️  This is a bug currently because this will use the default page size and cap the results at 20 (confirmed on upstream forks) .We want to fix that and the timeout (default is infinite, so we can set to 5 minutes or the like)

The best Route of action will probably be to use `ResultIterator` and return back as many results as possible. After importing it the only change needed should be to just something like this
```java
Filter filter = ...
return resultIterable(catalogFramework, new QueryRequestImpl(new QueryImpl(
    filter,
    ... all the other args including timeout,)
    .stream()
    .collect(Collectors.toList());
```

Reply:
i made the change, please double check it, the code is now in NoteUtil